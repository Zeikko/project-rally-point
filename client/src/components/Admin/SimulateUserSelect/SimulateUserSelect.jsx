import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { simulateUserAction } from '../../../actions/user-actions'

class SimulateUserSelect extends Component {
  constructor() {
    super()
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(event) {
    const { dispatch } = this.props
    dispatch(simulateUserAction(parseInt(event.target.value, 10)))
  }

  render() {
    let simulatedUserIds = []
    for (let id = 1; id <= 48; id += 1) {
      simulatedUserIds = [
        ...simulatedUserIds,
        id,
      ]
    }
    return (
      <div>
        <select onChange={event => this.handleChange(event)}>
          {simulatedUserIds.map(id =>
            <option key={id} value={id}>Test user {id}</option>)}
        </select>
      </div>
    )
  }
}

SimulateUserSelect.propTypes = {
  dispatch: PropTypes.func.isRequired,
}

export default SimulateUserSelect
