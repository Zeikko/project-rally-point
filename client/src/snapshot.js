import renderer from 'react-test-renderer'
import 'jest-styled-components'

export function matchComponentToSnapshot(component) {
  const json = renderer.create(component).toJSON()
  expect(json).toMatchSnapshot()
}