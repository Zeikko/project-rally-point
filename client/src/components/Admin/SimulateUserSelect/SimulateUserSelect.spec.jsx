import React from 'react'
import { mount } from 'enzyme'
import { matchComponentToSnapshot } from '../../../test/snapshot'
import SimulateUserSelect from './SimulateUserSelect'

describe('SimulateUserSelect', () => {
  it('renders correctly', () => {
    matchComponentToSnapshot(<SimulateUserSelect dispatch={() => {}} />)
  })

  it('joins a game', () => {
    const dispatch = jest.fn()
    const wrapper = mount(<SimulateUserSelect dispatch={dispatch} />)
    wrapper.find('select').simulate('change')
    expect(dispatch.mock.calls[0]).toEqual([
      {
        type: 'SIMULATE_USER',
        id: 1,
      },
    ])
  })
})
