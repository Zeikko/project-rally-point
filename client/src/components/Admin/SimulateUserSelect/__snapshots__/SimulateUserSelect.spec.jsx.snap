// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulateUserSelect renders correctly 1`] = `
<div>
  <select
    onChange={[Function]}
  >
    <option
      value={1}
    >
      Test user 
      1
    </option>
    <option
      value={2}
    >
      Test user 
      2
    </option>
    <option
      value={3}
    >
      Test user 
      3
    </option>
    <option
      value={4}
    >
      Test user 
      4
    </option>
    <option
      value={5}
    >
      Test user 
      5
    </option>
    <option
      value={6}
    >
      Test user 
      6
    </option>
    <option
      value={7}
    >
      Test user 
      7
    </option>
    <option
      value={8}
    >
      Test user 
      8
    </option>
    <option
      value={9}
    >
      Test user 
      9
    </option>
    <option
      value={10}
    >
      Test user 
      10
    </option>
    <option
      value={11}
    >
      Test user 
      11
    </option>
    <option
      value={12}
    >
      Test user 
      12
    </option>
    <option
      value={13}
    >
      Test user 
      13
    </option>
    <option
      value={14}
    >
      Test user 
      14
    </option>
    <option
      value={15}
    >
      Test user 
      15
    </option>
    <option
      value={16}
    >
      Test user 
      16
    </option>
    <option
      value={17}
    >
      Test user 
      17
    </option>
    <option
      value={18}
    >
      Test user 
      18
    </option>
    <option
      value={19}
    >
      Test user 
      19
    </option>
    <option
      value={20}
    >
      Test user 
      20
    </option>
    <option
      value={21}
    >
      Test user 
      21
    </option>
    <option
      value={22}
    >
      Test user 
      22
    </option>
    <option
      value={23}
    >
      Test user 
      23
    </option>
    <option
      value={24}
    >
      Test user 
      24
    </option>
    <option
      value={25}
    >
      Test user 
      25
    </option>
    <option
      value={26}
    >
      Test user 
      26
    </option>
    <option
      value={27}
    >
      Test user 
      27
    </option>
    <option
      value={28}
    >
      Test user 
      28
    </option>
    <option
      value={29}
    >
      Test user 
      29
    </option>
    <option
      value={30}
    >
      Test user 
      30
    </option>
    <option
      value={31}
    >
      Test user 
      31
    </option>
    <option
      value={32}
    >
      Test user 
      32
    </option>
    <option
      value={33}
    >
      Test user 
      33
    </option>
    <option
      value={34}
    >
      Test user 
      34
    </option>
    <option
      value={35}
    >
      Test user 
      35
    </option>
    <option
      value={36}
    >
      Test user 
      36
    </option>
    <option
      value={37}
    >
      Test user 
      37
    </option>
    <option
      value={38}
    >
      Test user 
      38
    </option>
    <option
      value={39}
    >
      Test user 
      39
    </option>
    <option
      value={40}
    >
      Test user 
      40
    </option>
    <option
      value={41}
    >
      Test user 
      41
    </option>
    <option
      value={42}
    >
      Test user 
      42
    </option>
    <option
      value={43}
    >
      Test user 
      43
    </option>
    <option
      value={44}
    >
      Test user 
      44
    </option>
    <option
      value={45}
    >
      Test user 
      45
    </option>
    <option
      value={46}
    >
      Test user 
      46
    </option>
    <option
      value={47}
    >
      Test user 
      47
    </option>
    <option
      value={48}
    >
      Test user 
      48
    </option>
  </select>
</div>
`;
